src/main/java/games/strategy/triplea/delegate/MustFightBattle.java

```diff
@@ -1201,7 +1203,7 @@ private boolean canAttackerRetreatPartialAmphib() {
     // non-paratrooper non-amphibious
     // land units.
     // If attacker is all planes, just return collection of current territory
-    if (m_headless || Match.allMatchNotEmpty(m_attackingUnits, Matches.UnitIsAir)
+    if (m_headless || !m_attackingUnits.isEmpty() && Match.allMatch(m_attackingUnits, Matches.UnitIsAir)
```

Comment:
Parentheses are not required here, but it may make it easier to read due to the `||`.